x/rewards/keeper/providers.go

```diff
@@ -39,40 +51,68 @@ func (k Keeper) distributeMonthlyBonusRewards(ctx sdk.Context) {
 		k.removeAllBasePay(ctx)
 		utils.LogLavaEvent(ctx, k.Logger(ctx), types.ProvidersBonusRewardsEventName, details, "provider bonus rewards distributed successfully")
 	}()
+
+	// Get serviced CU for each provider + spec
+	specCuMap := map[string]types.SpecCuType{} // spec -> specCu
 	for _, spec := range specs {
 		// all providers basepays and the total basepay of the spec
 		basepays, totalbasepay := k.specProvidersBasePay(ctx, spec.ChainID)
-		if totalbasepay.IsZero() {
+		if len(basepays) == 0 {
```

Comment:
why did you do this change?

Reply:
because if a provider provides service for a consumer for free (free sub), the total could be zero but there is iprpc cu to account for. So checking if the basepays list is more accurate when dealing with both provider bonus rewards and iprpc

Reply:
ahh nice.